### PR TITLE
Add check for running from current user directory and bail with warning

### PR DESCRIPTION
**Pull request by natewalck**
This isn't perfect, but it'll help dumb people like me from running it in my home directory and wondering why it doesn't work 😄 

Run from my Desktop:
```
sudo ./installinstallmacos.py
Running this script from /Users/nate/Desktop may not work as expected. If this does not run as expected, Please run again from somewhere else, such as /Users/Shared
Downloading https://swscan.apple.com/content/catalogs/others/index-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog...
...
```

Run from ~/code/macadmin-scripts:

```
sudo ./installinstallmacos.py
Downloading https://swscan.apple.com/content/catalogs/others/index-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog...
...
```

### CODE REVIEW

File: installinstallmacos.py [hunk 1]
```diff
@@ -511,8 +511,8 @@ def main():
         bad_dirs = ['Documents', 'Desktop', 'Downloads', 'Library']
         for bad_dir in bad_dirs:
             if bad_dir in current_dir:
-                sys.exit('This command cannot be run from within certain home directories, so please '
-                'run again from somewhere else, such as /Users/Shared')
+                print('Running this script from {} may not work as expected. If this does not run as expected, '
+                'please run again from somewhere else, such as /Users/Shared'.format(current_dir))
```
Comment (gregneagle) [inline installinstallmacos.py:516]: print this warning to stderr, please

Comment (gregneagle) [reply installinstallmacos.py:516]: Second line indentation looks wrong

Comment (gregneagle) [reply installinstallmacos.py:516]: The script doesn't use .format()-style string interpolation anywhere else. Should stick to the established style.

Comment (gregneagle) [reply installinstallmacos.py:516]: Also: what happens if it's run from `/Users/Shared/Documents`?

Comment (natewalck) [reply installinstallmacos.py:516]: It is fine in `/Users/Shared/Documents` as it won't pass the first check on line 510 (which is looking for user dir). So we check for user dir first, then get more specific after that.

```
 sudo ./installinstallmacos.py
Password:
Downloading https://swscan.apple.com/content/catalogs/others/index-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog...
...
```

File: installinstallmacos.py [hunk 2]
```diff
@@ -506,6 +506,16 @@ def main():
...
 
+    current_dir = os.getcwd()
+    if os.path.expanduser("~") in current_dir:
+        bad_dirs = ['Documents', 'Desktop', 'Downloads', 'Library']
+        for bad_dir in bad_dirs:
+            if bad_dir in current_dir:
+                print('Running this script from %s may not work as expected. '
+                'If this does not run as expected, please run again from '
+                'somewhere else, such as /Users/Shared.'
+                % current_dir, file=sys.stderr)
```
Comment (gregneagle) [inline installinstallmacos.py:517]: This indentation is still wrong. It should be:
```python
                print('Running this script from %s may not work as expected. '
                      'If this does not run as expected, please run again from '
                      'somewhere else, such as /Users/Shared.'
                      % current_dir, file=sys.stderr)
```

Comment (natewalck) [reply installinstallmacos.py]: Done. I was thinking line length for some reason.

File: installinstallmacos.py [hunk 3]
```diff
@@ -506,6 +506,16 @@ def main():
         sys.exit('This command requires root (to install packages), so please '
                  'run again with sudo or as root.')
 
+    current_dir = os.getcwd()
+    if os.path.expanduser("~") in current_dir:
+        bad_dirs = ['Documents', 'Desktop', 'Downloads', 'Library']
+        for bad_dir in bad_dirs:
+            if bad_dir in current_dir:
```
Comment (gregneagle) [inline installinstallmacos.py:513]: Safer to do `if bad_dir in os.path.split(current_dir): ` so you don't get false matches against `/Users/foo/MovedFromDesktop`

Comment (gregneagle) [reply installinstallmacos.py:513]:
```
>>> this_dir = "/Users/foo/MovedFromDesktop"
>>> "Desktop" in this_dir
True
>>> "Desktop" in os.path.split(this_dir)
False
```

File: installinstallmacos.py [hunk 4]
```diff
@@ -506,6 +506,16 @@ def main():
...
+                print('Running this script from %s may not work as expected. '
+                    'If this does not run as expected, please run again from '
+                    'somewhere else, such as /Users/Shared.'
+                    % current_dir, file=sys.stderr)
```
Comment (gregneagle) [inline installinstallmacos.py:517]: Indent _still_ wrong...

Comment (natewalck) [reply installinstallmacos.py]: I hate python indents.

**Comment by grahampugh** [comment]
@natewalck I think there is a mistake - you have:
```
if bad_dir in os.path.split(this_dir):
```
But I think you mean:
```
if bad_dir in os.path.split(current_dir):
```

**Comment by gregneagle** [comment]
Fixed it.

**Comment by natewalck** [comment]
Yep, bad copy pasta from your example. I had pasted in the current_dir, but must have lost it somehow before I diffed.

**Comment by grahampugh** [comment]
`black` was my friend, letting me know there was an unused variable. 
It also disagrees with the indentation of the print statement, asserting this instead:
```
                print(
                    "Running this script from %s may not work as expected. "
                    "If this does not run as expected, please run again from "
                    "somewhere else, such as /Users/Shared." % current_dir,
                    file=sys.stderr,
                )
```
But that's just opinion...

**Comment by natewalck** [comment]
I should just run black on save :D I've gotten so used to that with Go, I really miss it.

**Comment by gregneagle** [comment]
I'm not using black, I use pylint, and so the code formatting might differ.

**Pull request merged**